{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "6479e3c35faa4031a7ae0b570727bffb",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/11 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from data_processing.clinvar import getClinvar\n",
    "from data_processing.main import get_gene_info, read_scoreset\n",
    "from data_processing.gnomad import queryGnomAD\n",
    "from data_processing.splice_ai import querySpliceAI\n",
    "from pathlib import Path\n",
    "import os\n",
    "from tqdm.autonotebook import tqdm\n",
    "import json\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import joblib\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from collections import defaultdict\n",
    "pd.set_option('display.max_columns', None)\n",
    "cache_dir = Path(\"/data/dzeiberg/mave_calibration/cache\")\n",
    "cache_dir.mkdir(exist_ok=True, parents=True)\n",
    "data_directory = Path(\"/data/dzeiberg/mave_calibration/data\")\n",
    "dataset_names = [os.path.basename(f.parent) for f in data_directory.glob(\"**/observations.pkl\")]\n",
    "metadata = []\n",
    "for dataset in dataset_names:\n",
    "    with open(data_directory / dataset / \"metadata.json\") as f:\n",
    "        metadata.append(dict(**json.load(f), dataset=dataset))\n",
    "metadata = pd.DataFrame(metadata).set_index(\"dataset\")\n",
    "gene_info = {}\n",
    "for uniprot_acc in tqdm(metadata[\"uniprot_acc\"].unique()):\n",
    "    gene_info[uniprot_acc] = get_gene_info(uniprot_acc, cache_dir=\"/data/dzeiberg/mave_calibration/cache\")\n",
    "gene_info = pd.DataFrame(gene_info).T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data_processing/config.json\") as f:\n",
    "    external_config = json.load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "read gnomAD missense variants in chromosomal regions of the transcripts for which I have MAVE data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "51fa16264a0d4297bd4d1c8dba0fc079",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "0it [00:00, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 17:43044295-43170245 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:27.458327.vcf\n",
      "12:04:28.982 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:29.013 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:29.015 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:29.015 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:29.016 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:29.016 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:29.016 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:28 PM EDT\n",
      "12:04:29.016 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:29.016 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:29.017 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:29.017 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:29.017 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:29.017 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:29.018 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:29.018 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:29.018 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:29.018 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:29.018 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:29.018 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:29.018 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:29.019 INFO  SelectVariants - Initializing engine\n",
      "12:04:29.110 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:29.212 INFO  IntervalArgumentCollection - Processing 125951 bp from intervals\n",
      "12:04:29.216 INFO  SelectVariants - Done initializing engine\n",
      "12:04:29.231 INFO  ProgressMeter - Starting traversal\n",
      "12:04:29.231 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:30.124 INFO  ProgressMeter -          17:43125295              0.0                239562       16168413.9\n",
      "12:04:30.125 INFO  ProgressMeter - Traversal complete. Processed 239562 total variants in 0.0 minutes.\n",
      "12:04:30.145 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:30 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 17:7661779-7687546 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:32.586411.vcf\n",
      "12:04:34.150 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:34.181 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:34.183 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:34.183 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:34.184 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:34.184 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:34.184 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:34 PM EDT\n",
      "12:04:34.184 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:34.184 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:34.185 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:34.185 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:34.185 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:34.185 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:34.185 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:34.185 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:34.186 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:34.186 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:34.186 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:34.186 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:34.186 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:34.186 INFO  SelectVariants - Initializing engine\n",
      "12:04:34.270 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:34.369 INFO  IntervalArgumentCollection - Processing 25768 bp from intervals\n",
      "12:04:34.372 INFO  SelectVariants - Done initializing engine\n",
      "12:04:34.387 INFO  ProgressMeter - Starting traversal\n",
      "12:04:34.387 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:34.764 INFO  ProgressMeter -           17:7687403              0.0                 61839        9947292.2\n",
      "12:04:34.764 INFO  ProgressMeter - Traversal complete. Processed 61839 total variants in 0.0 minutes.\n",
      "12:04:34.770 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:34 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.01 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 13:32315086-32400268 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:35.441902.vcf\n",
      "12:04:36.998 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:37.029 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:37.031 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:37.031 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:37.031 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:37.031 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:37.032 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:36 PM EDT\n",
      "12:04:37.032 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:37.032 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:37.033 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:37.033 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:37.033 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:37.033 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:37.033 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:37.034 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:37.034 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:37.034 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:37.034 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:37.034 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:37.034 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:37.035 INFO  SelectVariants - Initializing engine\n",
      "12:04:37.119 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:37.215 INFO  IntervalArgumentCollection - Processing 85183 bp from intervals\n",
      "12:04:37.219 INFO  SelectVariants - Done initializing engine\n",
      "12:04:37.232 INFO  ProgressMeter - Starting traversal\n",
      "12:04:37.233 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:38.191 INFO  ProgressMeter -          13:32400215              0.0                255150       16047169.8\n",
      "12:04:38.191 INFO  ProgressMeter - Traversal complete. Processed 255150 total variants in 0.0 minutes.\n",
      "12:04:38.207 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:38 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 18:23506184-23586506 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:39.956017.vcf\n",
      "12:04:41.494 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:41.522 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:41.524 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:41.524 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:41.525 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:41.525 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:41.525 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:41 PM EDT\n",
      "12:04:41.525 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:41.525 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:41.526 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:41.526 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:41.526 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:41.526 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:41.526 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:41.527 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:41.527 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:41.527 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:41.527 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:41.527 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:41.527 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:41.528 INFO  SelectVariants - Initializing engine\n",
      "12:04:41.608 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:41.708 INFO  IntervalArgumentCollection - Processing 80323 bp from intervals\n",
      "12:04:41.711 INFO  SelectVariants - Done initializing engine\n",
      "12:04:41.725 INFO  ProgressMeter - Starting traversal\n",
      "12:04:41.725 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:42.639 INFO  ProgressMeter -          18:23586504              0.0                242007       15938990.1\n",
      "12:04:42.640 INFO  ProgressMeter - Traversal complete. Processed 242007 total variants in 0.0 minutes.\n",
      "12:04:42.653 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:42 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 3:10141778-10153667 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:43.873408.vcf\n",
      "12:04:45.495 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:45.526 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:45.529 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:45.529 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:45.529 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:45.529 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:45.529 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:45 PM EDT\n",
      "12:04:45.529 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:45.529 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:45.530 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:45.530 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:45.530 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:45.531 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:45.531 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:45.531 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:45.531 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:45.531 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:45.531 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:45.531 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:45.532 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:45.532 INFO  SelectVariants - Initializing engine\n",
      "12:04:45.617 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:45.719 INFO  IntervalArgumentCollection - Processing 11890 bp from intervals\n",
      "12:04:45.723 INFO  SelectVariants - Done initializing engine\n",
      "12:04:45.737 INFO  ProgressMeter - Starting traversal\n",
      "12:04:45.738 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:45.977 INFO  ProgressMeter -           3:10152111              0.0                 31326        7964237.3\n",
      "12:04:45.977 INFO  ProgressMeter - Traversal complete. Processed 31326 total variants in 0.0 minutes.\n",
      "12:04:45.984 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:45 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.01 minutes.\n",
      "Runtime.totalMemory()=285212672\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 22:28687743-28742422 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:47.305243.vcf\n",
      "12:04:48.932 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:48.962 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:48.965 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:48.965 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:48.965 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:48.965 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:48.966 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:48 PM EDT\n",
      "12:04:48.966 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:48.966 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:48.967 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:48.967 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:48.967 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:48.967 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:48.967 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:48.967 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:48.968 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:48.968 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:48.968 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:48.968 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:48.968 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:48.968 INFO  SelectVariants - Initializing engine\n",
      "12:04:49.056 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:49.154 INFO  IntervalArgumentCollection - Processing 54680 bp from intervals\n",
      "12:04:49.157 INFO  SelectVariants - Done initializing engine\n",
      "12:04:49.172 INFO  ProgressMeter - Starting traversal\n",
      "12:04:49.172 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:49.821 INFO  ProgressMeter -          22:28742387              0.0                142107       13239782.6\n",
      "12:04:49.822 INFO  ProgressMeter - Traversal complete. Processed 142107 total variants in 0.0 minutes.\n",
      "12:04:49.836 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:49 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 2:47403067-47663146 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:50.871993.vcf\n",
      "12:04:52.426 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:52.455 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:52.458 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:52.458 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:52.458 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:52.458 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:52.458 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:52 PM EDT\n",
      "12:04:52.458 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:52.458 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:52.459 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:52.459 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:52.459 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:52.459 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:52.460 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:52.460 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:52.460 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:52.460 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:52.460 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:52.460 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:52.460 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:52.461 INFO  SelectVariants - Initializing engine\n",
      "12:04:52.540 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:52.637 INFO  IntervalArgumentCollection - Processing 260080 bp from intervals\n",
      "12:04:52.640 INFO  SelectVariants - Done initializing engine\n",
      "12:04:52.654 INFO  ProgressMeter - Starting traversal\n",
      "12:04:52.655 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:53.960 INFO  ProgressMeter -           2:47574609              0.0                410748       18928479.3\n",
      "12:04:53.961 INFO  ProgressMeter - Traversal complete. Processed 410748 total variants in 0.0 minutes.\n",
      "12:04:53.982 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:53 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.03 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 10:87862638-87971930 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:04:55.935033.vcf\n",
      "12:04:57.535 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:04:57.566 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:57.568 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:04:57.568 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:04:57.569 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:04:57.569 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:04:57.569 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:04:57 PM EDT\n",
      "12:04:57.569 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:57.569 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:04:57.570 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:04:57.570 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:04:57.570 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:04:57.570 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:04:57.570 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:04:57.571 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:04:57.571 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:04:57.571 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:04:57.571 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:04:57.571 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:04:57.571 INFO  SelectVariants - Requester pays: disabled\n",
      "12:04:57.572 INFO  SelectVariants - Initializing engine\n",
      "12:04:57.659 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:04:57.758 INFO  IntervalArgumentCollection - Processing 109293 bp from intervals\n",
      "12:04:57.761 INFO  SelectVariants - Done initializing engine\n",
      "12:04:57.775 INFO  ProgressMeter - Starting traversal\n",
      "12:04:57.775 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:04:58.922 INFO  ProgressMeter -          10:87971646              0.0                328851       17277635.7\n",
      "12:04:58.922 INFO  ProgressMeter - Traversal complete. Processed 328851 total variants in 0.0 minutes.\n",
      "12:04:58.960 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:04:58 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 6:18128311-18155077 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:05:00.562173.vcf\n",
      "12:05:02.149 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:05:02.182 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:02.184 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:05:02.185 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:05:02.185 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:05:02.185 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:05:02.185 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:05:02 PM EDT\n",
      "12:05:02.185 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:02.185 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:02.186 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:05:02.186 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:05:02.186 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:05:02.187 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:05:02.187 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:05:02.187 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:05:02.187 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:05:02.187 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:05:02.187 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:05:02.187 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:05:02.188 INFO  SelectVariants - Requester pays: disabled\n",
      "12:05:02.188 INFO  SelectVariants - Initializing engine\n",
      "12:05:02.270 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:05:02.367 INFO  IntervalArgumentCollection - Processing 26767 bp from intervals\n",
      "12:05:02.370 INFO  SelectVariants - Done initializing engine\n",
      "12:05:02.386 INFO  ProgressMeter - Starting traversal\n",
      "12:05:02.387 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:05:02.802 INFO  ProgressMeter -           6:18154977              0.0                 80289       11721021.9\n",
      "12:05:02.802 INFO  ProgressMeter - Traversal complete. Processed 80289 total variants in 0.0 minutes.\n",
      "12:05:02.814 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:05:02 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.01 minutes.\n",
      "Runtime.totalMemory()=1224736768\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L X:41333348-41364472 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:05:03.746785.vcf\n",
      "12:05:05.360 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:05:05.391 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:05.394 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:05:05.394 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:05:05.394 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:05:05.394 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:05:05.394 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:05:05 PM EDT\n",
      "12:05:05.395 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:05.395 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:05.395 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:05:05.395 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:05:05.396 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:05:05.396 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:05:05.396 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:05:05.396 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:05:05.396 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:05:05.396 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:05:05.397 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:05:05.397 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:05:05.397 INFO  SelectVariants - Requester pays: disabled\n",
      "12:05:05.397 INFO  SelectVariants - Initializing engine\n",
      "12:05:05.483 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:05:05.580 INFO  IntervalArgumentCollection - Processing 31125 bp from intervals\n",
      "12:05:05.583 INFO  SelectVariants - Done initializing engine\n",
      "12:05:05.597 INFO  ProgressMeter - Starting traversal\n",
      "12:05:05.598 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:05:05.929 INFO  ProgressMeter -           X:41350014              0.0                 50778        9345644.2\n",
      "12:05:05.929 INFO  ProgressMeter - Traversal complete. Processed 50778 total variants in 0.0 minutes.\n",
      "12:05:05.949 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:05:05 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.01 minutes.\n",
      "Runtime.totalMemory()=285212672\n",
      "Using GATK jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar SelectVariants -V /data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz -L 11:119084866-119093834 --output /data/dzeiberg/mave_calibration/cache/splice_ai_query_result.2024-09-12_12:05:06.967244.vcf\n",
      "12:05:08.591 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/data/utilities/bio/gatk-4.4.0.0/gatk-package-4.4.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "12:05:08.624 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:08.627 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.4.0.0\n",
      "12:05:08.627 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "12:05:08.627 INFO  SelectVariants - Executing as dzeiberg@thebigticket.khoury.northeastern.edu on Linux v3.10.0-1160.36.2.el7.x86_64 amd64\n",
      "12:05:08.627 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v18.0.1+10\n",
      "12:05:08.628 INFO  SelectVariants - Start Date/Time: September 12, 2024 at 12:05:08 PM EDT\n",
      "12:05:08.628 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:08.628 INFO  SelectVariants - ------------------------------------------------------------\n",
      "12:05:08.628 INFO  SelectVariants - HTSJDK Version: 3.0.5\n",
      "12:05:08.629 INFO  SelectVariants - Picard Version: 3.0.0\n",
      "12:05:08.629 INFO  SelectVariants - Built for Spark Version: 3.3.1\n",
      "12:05:08.629 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "12:05:08.629 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "12:05:08.629 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "12:05:08.630 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "12:05:08.630 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "12:05:08.630 INFO  SelectVariants - Inflater: IntelInflater\n",
      "12:05:08.630 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "12:05:08.630 INFO  SelectVariants - Requester pays: disabled\n",
      "12:05:08.630 INFO  SelectVariants - Initializing engine\n",
      "12:05:08.718 INFO  FeatureManager - Using codec VCFCodec to read file file:///data/dbs/spliceAI/genome_scores_v1.3_ds.20a701bc58ab45b59de2576db79ac8d0/spliceai_scores.raw.snv.hg38.vcf.gz\n",
      "12:05:08.819 INFO  IntervalArgumentCollection - Processing 8969 bp from intervals\n",
      "12:05:08.822 INFO  SelectVariants - Done initializing engine\n",
      "12:05:08.837 INFO  ProgressMeter - Starting traversal\n",
      "12:05:08.837 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "12:05:09.100 INFO  ProgressMeter -         11:119093216              0.0                 25992        6068171.2\n",
      "12:05:09.100 INFO  ProgressMeter - Traversal complete. Processed 25992 total variants in 0.0 minutes.\n",
      "12:05:09.148 INFO  SelectVariants - Shutting down engine\n",
      "[September 12, 2024 at 12:05:09 PM EDT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.01 minutes.\n",
      "Runtime.totalMemory()=285212672\n"
     ]
    }
   ],
   "source": [
    "gnomad_records = []\n",
    "spliceAI_scores = []\n",
    "for uniprot_acc, r in tqdm(gene_info.iterrows()):\n",
    "    if (cache_dir / f\"gnomad_matches_{r.HGNC_ID}_CHR{r.CHROM}_{r.START}_{r.STOP}.tsv\").exists():\n",
    "        gnomad_record = pd.read_csv(cache_dir / f\"gnomad_matches_{r.HGNC_ID}_CHR{r.CHROM}_{r.START}_{r.STOP}.tsv\", sep=\"\\t\")\n",
    "    else:\n",
    "        print(cache_dir / f\"gnomad_matches_{r.HGNC_ID}_CHR{r.CHROM}_{r.START}_{r.STOP}.tsv\")\n",
    "        break\n",
    "        gnomad_record = queryGnomAD(r.CHROM, r.START,r.STOP,r.HGNC_ID,\n",
    "                                            **external_config[\"gnomad\"],\n",
    "                                            **external_config['external_tools'],\n",
    "                                            write_dir=cache_dir)\n",
    "    gnomad_record = gnomad_record.assign(CHROM=gnomad_record.CHROM.astype(str).str.replace('chr',''),\n",
    "                            POS=gnomad_record.POS.astype(str),\n",
    "                            REF=gnomad_record.REF.astype(str),\n",
    "                            ALT=gnomad_record.ALT.astype(str))\n",
    "    # read spliceAI scores\n",
    "    gnomad_records.append(gnomad_record)\n",
    "    hg38_fp = cache_dir / f\"splice_ai_hg38_match_CHR{r.CHROM}_{r.START}_{r.STOP}.tsv\"\n",
    "    if hg38_fp.exists():\n",
    "        region_spliceAI_scores_hg38 = pd.read_csv(hg38_fp, sep=\"\\t\")\n",
    "    else:\n",
    "        region_spliceAI_scores_hg38 = querySpliceAI(r['CHROM'],r['START'],r['STOP'],spliceAIFilePath=external_config['splice_ai']['spliceAIFilePath_hg38'],assembly='hg38',write_dir=cache_dir)\n",
    "    region_spliceAI_scores_hg38 = region_spliceAI_scores_hg38.assign(CHROM=region_spliceAI_scores_hg38.CHROM.astype(str),\n",
    "                                                           POS=region_spliceAI_scores_hg38.POS.astype(str),\n",
    "                                                           REF=region_spliceAI_scores_hg38.REF.astype(str),\n",
    "                                                              ALT=region_spliceAI_scores_hg38.ALT.astype(str),\n",
    "                                                              Assembly=\"GRCh38\")\n",
    "    spliceAI_scores.append(region_spliceAI_scores_hg38)\n",
    "# aggregate gnomAD data    \n",
    "gnomad_df = pd.concat(gnomad_records)\n",
    "gnomad_df = gnomad_df.assign(Feature_base=gnomad_df.Feature.str.split(\".\").str[0])\n",
    "# aggregate spliceAI scores\n",
    "spliceAI_df = pd.concat(spliceAI_scores).set_index(['CHROM','POS','REF','ALT'],)\n",
    "spliceAI_scoreDict = spliceAI_df.spliceAI_score.to_dict()\n",
    "gnomad_count = gnomad_df.shape[0]\n",
    "gnomad_df = gnomad_df.set_index([\"CHROM\",\"POS\",\"REF\",\"ALT\"]).assign(spliceAI_scores=spliceAI_scoreDict)\n",
    "assert gnomad_count == gnomad_df.shape[0]\n",
    "# Read ClinVar data\n",
    "clinvar_df = getClinvar(cache_dir=cache_dir, use_cached_processed_file=True)\n",
    "clinvar_df = clinvar_df.assign(transcript_base=clinvar_df.transcript.str.split(\".\").str[0])\n",
    "# only variants with a base transcript in gene_info (only relevant variants)\n",
    "clinvar_df[clinvar_df.transcript_base.isin(set(gene_info.MANE_RefSeq_nuc.str.split(\".\").str[0].values))]\n",
    "# only GRCh38 Assembly variants\n",
    "clinvar_df = clinvar_df[clinvar_df.Assembly == \"GRCh38\"]\n",
    "# annotate with spliceAI scores\n",
    "\n",
    "clinvar_df = clinvar_df.assign(Chromosome=clinvar_df.Chromosome.astype(str),\n",
    "                                 PositionVCF=clinvar_df.PositionVCF.astype(str))\n",
    "clinvar_count = clinvar_df.shape[0]\n",
    "clinvar_df = clinvar_df.set_index([\"Chromosome\",\"PositionVCF\",\"ReferenceAlleleVCF\",\"AlternateAlleleVCF\"]).assign(spliceAI_score=spliceAI_scoreDict)\n",
    "assert clinvar_count == clinvar_df.shape[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/dzeiberg/mave_calibration/cache/mapping_data.pkl']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "joblib.dump(dict(metadata=metadata, gene_info=gene_info, gnomad_df=gnomad_df, spliceAI_df=spliceAI_df, clinvar_df=clinvar_df), \"/data/dzeiberg/mave_calibration/cache/mapping_data.pkl\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "counts of gnomAD variants for each gene of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SYMBOL\n",
       "BRCA2    5103\n",
       "BRCA1    2538\n",
       "NPC1     1914\n",
       "MSH2     1852\n",
       "CHEK2     859\n",
       "TP53      605\n",
       "HMBS      494\n",
       "VHL       405\n",
       "TPMT      336\n",
       "DDX3X     268\n",
       "PTEN      254\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomad_df[gnomad_df.Feature_base.isin(set(gene_info.MANE_RefSeq_nuc.str.split(\".\").str[0].values))].SYMBOL.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "inverting Boettcher_TP53 scores\n",
      "skipping Erwood_NPC1_RPE1\n",
      "inverting Giacomelli_p53_nullNutlin3 scores\n",
      "inverting Giacomelli_p53_wtNutlin3 scores\n",
      "inverting Jia_MSH2_SSM scores\n",
      "standardizing Kato_TP53_AIP1nWT to b_lb\n",
      "standardizing Kato_TP53_BAXnWT to b_lb\n",
      "standardizing Kato_TP53_GADD45nWT to b_lb\n",
      "standardizing Kato_TP53_MDM2nWT to b_lb\n",
      "standardizing Kato_TP53_NOXAnWT to b_lb\n",
      "standardizing Kato_TP53_P53R2nWT to b_lb\n",
      "standardizing Kato_TP53_WAF1nWT to b_lb\n",
      "standardizing Kato_TP53_h1433snWT to b_lb\n",
      "inverting Kotler_TP53_RelativeFitnessH1299 scores\n",
      "inverting Kotler_TP53_RelativeFitnessHCT116 scores\n",
      "skipping Kotler_TP53_inVivoEnrichment\n",
      "skipping Matreyek_TPMT_VampSeq\n"
     ]
    }
   ],
   "source": [
    "data_directory = Path(\"/data/dzeiberg/mave_calibration/data\")\n",
    "with open(data_directory / \"dataset_configs.json\") as f:\n",
    "    dataset_configs = json.load(f)\n",
    "scoresets = dict()\n",
    "for dataset_name in dataset_names:\n",
    "    if dataset_name not in dataset_configs:\n",
    "        print(f\"skipping {dataset_name}\")\n",
    "        continue\n",
    "    # read raw score\n",
    "    scoreset = read_scoreset(data_directory / dataset_name / \"scoreset.csv\")\n",
    "    # standardize if necessary\n",
    "    standardize_sample_name = dataset_configs[dataset_name].get('standardize_to',None)\n",
    "    if standardize_sample_name is not None:\n",
    "        print(f'standardizing {dataset_name} to {standardize_sample_name}')\n",
    "        standardization_sample = joblib.load(data_directory / dataset_name / \"observations.pkl\")[standardize_sample_name]\n",
    "        mu = standardization_sample.mean()\n",
    "        sigma = standardization_sample.std()\n",
    "        scoreset = scoreset.assign(score= (scoreset.score - mu) / sigma)\n",
    "    # invert if necessary\n",
    "    if dataset_configs[dataset_name].get('invert',False):\n",
    "        print(f'inverting {dataset_name} scores')\n",
    "        scoreset = scoreset.assign(score=-1 * scoreset.score)\n",
    "    uniprot_acc = metadata.loc[dataset_name,\"uniprot_acc\"]\n",
    "    # add metadata\n",
    "    scoreset = scoreset.assign(GeneID=int(gene_info.loc[uniprot_acc,'GeneID']),\n",
    "                                MANE_RefSeq_nuc=gene_info.loc[uniprot_acc,'MANE_RefSeq_nuc'],\n",
    "                                MANE_RefSeq_nuc_base=gene_info.loc[uniprot_acc,'MANE_RefSeq_nuc'].split('.')[0],\n",
    "                                HGNC_ID=gene_info.loc[uniprot_acc,'HGNC_ID'],\n",
    "                                Ensembl_nuc=gene_info.loc[uniprot_acc,'ENSEMBL_nuc'],\n",
    "                                Ensembl_prot=gene_info.loc[uniprot_acc,'ENSEMBL_prot'],\n",
    "                                Ensembl_nuc_base=gene_info.loc[uniprot_acc,'ENSEMBL_nuc'].split('.')[0],\n",
    "                                Ensembl_prot_base=gene_info.loc[uniprot_acc,'ENSEMBL_prot'].split('.')[0])\n",
    "    scoresets[dataset_name] = scoreset\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gnomAD_subset(ss):\n",
    "    scoreset_tups_ensembl = set(ss.loc[:,['Ensembl_nuc_base','hgvs_pro']].astype(str).apply(tuple,axis=1).values)\n",
    "    scoreset_tups_refseq = set(ss.loc[:,['MANE_RefSeq_nuc_base','hgvs_pro']].astype(str).apply(tuple,axis=1).values)\n",
    "    # get gnomAD variants with available scores\n",
    "    gnomad_sub = gnomad_df[gnomad_df.loc[:,['Feature_base','hgvs_pro']].astype(str).apply(tuple,axis=1).isin(scoreset_tups_ensembl.union(scoreset_tups_refseq))].reset_index().set_index(['Feature_base','hgvs_pro'])\n",
    "    scores_ensembl = ss.set_index(['Ensembl_nuc_base','hgvs_pro']).score\n",
    "    scores_refseq = ss.set_index(['MANE_RefSeq_nuc_base','hgvs_pro']).score\n",
    "    gnomad_sub['score'] = np.nan\n",
    "    gnomad_sub['score'].update(scores_ensembl)# = gnomad_sub.assign(score=gnomad_sub.index.map(scores_ensembl)).reset_index()\n",
    "    gnomad_sub['score'].update(scores_refseq)\n",
    "    return gnomad_sub.reset_index()\n",
    "\n",
    "def get_clinvar_subset(ss):\n",
    "    scoreset_tups = set(ss.loc[:,['MANE_RefSeq_nuc_base','hgvs_pro']].astype(str).apply(tuple,axis=1).values)\n",
    "    clinvar_sub = clinvar_df[clinvar_df.loc[:,['transcript_base','hgvs_pro']].astype(str).apply(tuple,axis=1).isin(scoreset_tups)].reset_index().set_index(['transcript_base','hgvs_pro'])\n",
    "    scores = ss.set_index(['MANE_RefSeq_nuc_base','hgvs_pro']).score\n",
    "    clinvar_sub['score'] = np.nan\n",
    "    clinvar_sub['score'].update(scores)\n",
    "    return clinvar_sub.reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnomAD_ds = get_gnomAD_subset(scoresets['Findlay_BRCA1_SGE'])\n",
    "clinvar_ds = get_clinvar_subset(scoresets['Findlay_BRCA1_SGE'])\n",
    "scoreset = scoresets['Findlay_BRCA1_SGE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Feature_base</th>\n",
       "      <th>hgvs_pro</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ID</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "      <th>vep</th>\n",
       "      <th>index</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE_SELECT</th>\n",
       "      <th>MANE_PLUS_CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>UNIPROT_ISOFORM</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>LoF</th>\n",
       "      <th>LoF_filter</th>\n",
       "      <th>LoF_flags</th>\n",
       "      <th>LoF_info</th>\n",
       "      <th>spliceAI_scores</th>\n",
       "      <th>score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>p.Ala1669Ser</td>\n",
       "      <td>17</td>\n",
       "      <td>43067677</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs80357087</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>28</td>\n",
       "      <td>1.841770e-04</td>\n",
       "      <td>A|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>6068</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>16/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000357654.9:c.5005G&gt;T</td>\n",
       "      <td>ENSP00000350283.3:p.Ala1669Ser</td>\n",
       "      <td>5118</td>\n",
       "      <td>5005</td>\n",
       "      <td>1669</td>\n",
       "      <td>A/S</td>\n",
       "      <td>Gcc/Tcc</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>A2</td>\n",
       "      <td>CCDS11453.1</td>\n",
       "      <td>ENSP00000350283</td>\n",
       "      <td>P38398-1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>ENSP_mappings:1jnx&amp;ENSP_mappings:1n5o&amp;ENSP_map...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.401851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447</th>\n",
       "      <td>NM_007294</td>\n",
       "      <td>p.Ala1669Ser</td>\n",
       "      <td>17</td>\n",
       "      <td>43067677</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs80357087</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>28</td>\n",
       "      <td>1.841770e-04</td>\n",
       "      <td>A|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>6068</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>672</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>16/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_007294.4:c.5005G&gt;T</td>\n",
       "      <td>NP_009225.1:p.Ala1669Ser</td>\n",
       "      <td>5118</td>\n",
       "      <td>5005</td>\n",
       "      <td>1669</td>\n",
       "      <td>A/S</td>\n",
       "      <td>Gcc/Tcc</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENST00000357654.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009225.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.401851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448</th>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>p.Ala1669Ser</td>\n",
       "      <td>17</td>\n",
       "      <td>43067677</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs80357087</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>122</td>\n",
       "      <td>8.353400e-05</td>\n",
       "      <td>A|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>6069</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>16/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000357654.9:c.5005G&gt;T</td>\n",
       "      <td>ENSP00000350283.3:p.Ala1669Ser</td>\n",
       "      <td>5118</td>\n",
       "      <td>5005</td>\n",
       "      <td>1669</td>\n",
       "      <td>A/S</td>\n",
       "      <td>Gcc/Tcc</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>A2</td>\n",
       "      <td>CCDS11453.1</td>\n",
       "      <td>ENSP00000350283</td>\n",
       "      <td>P38398-1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>ENSP_mappings:1jnx&amp;ENSP_mappings:1n5o&amp;ENSP_map...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.401851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>NM_007294</td>\n",
       "      <td>p.Ala1669Ser</td>\n",
       "      <td>17</td>\n",
       "      <td>43067677</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs80357087</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>122</td>\n",
       "      <td>8.353400e-05</td>\n",
       "      <td>A|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>6069</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>672</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>16/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_007294.4:c.5005G&gt;T</td>\n",
       "      <td>NP_009225.1:p.Ala1669Ser</td>\n",
       "      <td>5118</td>\n",
       "      <td>5005</td>\n",
       "      <td>1669</td>\n",
       "      <td>A/S</td>\n",
       "      <td>Gcc/Tcc</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENST00000357654.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009225.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.401851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451</th>\n",
       "      <td>NM_007294</td>\n",
       "      <td>p.Ala1669Thr</td>\n",
       "      <td>17</td>\n",
       "      <td>43067677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs80357087</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>6.847040e-07</td>\n",
       "      <td>T|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>6070</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>672</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>16/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_007294.4:c.5005G&gt;A</td>\n",
       "      <td>NP_009225.1:p.Ala1669Thr</td>\n",
       "      <td>5118</td>\n",
       "      <td>5005</td>\n",
       "      <td>1669</td>\n",
       "      <td>A/T</td>\n",
       "      <td>Gcc/Acc</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENST00000357654.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009225.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.033971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>p.Val83Ile</td>\n",
       "      <td>17</td>\n",
       "      <td>43104922</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs1060502343</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>6</td>\n",
       "      <td>4.105090e-06</td>\n",
       "      <td>T|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>16841</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000357654.9:c.247G&gt;A</td>\n",
       "      <td>ENSP00000350283.3:p.Val83Ile</td>\n",
       "      <td>360</td>\n",
       "      <td>247</td>\n",
       "      <td>83</td>\n",
       "      <td>V/I</td>\n",
       "      <td>Gtt/Att</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>A2</td>\n",
       "      <td>CCDS11453.1</td>\n",
       "      <td>ENSP00000350283</td>\n",
       "      <td>P38398-1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>ENSP_mappings:1jm7&amp;Gene3D:3&amp;PIRSF:PIRSF001734&amp;...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.07</td>\n",
       "      <td>-0.093432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>777</th>\n",
       "      <td>NM_007294</td>\n",
       "      <td>p.Val8Ile</td>\n",
       "      <td>17</td>\n",
       "      <td>43124075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs528902306</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>6.843080e-07</td>\n",
       "      <td>T|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>21127</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>672</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_007294.4:c.22G&gt;A</td>\n",
       "      <td>NP_009225.1:p.Val8Ile</td>\n",
       "      <td>135</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>V/I</td>\n",
       "      <td>Gtt/Att</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENST00000357654.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009225.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.560919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>776</th>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>p.Val8Ile</td>\n",
       "      <td>17</td>\n",
       "      <td>43124075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs528902306</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>6.843080e-07</td>\n",
       "      <td>T|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>21127</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000357654.9:c.22G&gt;A</td>\n",
       "      <td>ENSP00000350283.3:p.Val8Ile</td>\n",
       "      <td>135</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>V/I</td>\n",
       "      <td>Gtt/Att</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>A2</td>\n",
       "      <td>CCDS11453.1</td>\n",
       "      <td>ENSP00000350283</td>\n",
       "      <td>P38398-1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>ENSP_mappings:1jm7&amp;Gene3D:3&amp;PIRSF:PIRSF001734&amp;...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.560919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>775</th>\n",
       "      <td>NM_007294</td>\n",
       "      <td>p.Val8Ile</td>\n",
       "      <td>17</td>\n",
       "      <td>43124075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs528902306</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>3</td>\n",
       "      <td>1.970130e-05</td>\n",
       "      <td>T|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>21126</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>672</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_007294.4:c.22G&gt;A</td>\n",
       "      <td>NP_009225.1:p.Val8Ile</td>\n",
       "      <td>135</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>V/I</td>\n",
       "      <td>Gtt/Att</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>ENST00000357654.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009225.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.560919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>774</th>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>p.Val8Ile</td>\n",
       "      <td>17</td>\n",
       "      <td>43124075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs528902306</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>3</td>\n",
       "      <td>1.970130e-05</td>\n",
       "      <td>T|missense_variant|MODERATE|BRCA1|ENSG00000012...</td>\n",
       "      <td>21126</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000357654.9:c.22G&gt;A</td>\n",
       "      <td>ENSP00000350283.3:p.Val8Ile</td>\n",
       "      <td>135</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>V/I</td>\n",
       "      <td>Gtt/Att</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>A2</td>\n",
       "      <td>CCDS11453.1</td>\n",
       "      <td>ENSP00000350283</td>\n",
       "      <td>P38398-1</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>ENSP_mappings:1jm7&amp;Gene3D:3&amp;PIRSF:PIRSF001734&amp;...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.560919</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>804 rows  61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Feature_base      hgvs_pro CHROM       POS REF ALT            ID  \\\n",
       "446  ENST00000357654  p.Ala1669Ser    17  43067677   C   A    rs80357087   \n",
       "447        NM_007294  p.Ala1669Ser    17  43067677   C   A    rs80357087   \n",
       "448  ENST00000357654  p.Ala1669Ser    17  43067677   C   A    rs80357087   \n",
       "449        NM_007294  p.Ala1669Ser    17  43067677   C   A    rs80357087   \n",
       "451        NM_007294  p.Ala1669Thr    17  43067677   C   T    rs80357087   \n",
       "..               ...           ...   ...       ...  ..  ..           ...   \n",
       "580  ENST00000357654    p.Val83Ile    17  43104922   C   T  rs1060502343   \n",
       "777        NM_007294     p.Val8Ile    17  43124075   C   T   rs528902306   \n",
       "776  ENST00000357654     p.Val8Ile    17  43124075   C   T   rs528902306   \n",
       "775        NM_007294     p.Val8Ile    17  43124075   C   T   rs528902306   \n",
       "774  ENST00000357654     p.Val8Ile    17  43124075   C   T   rs528902306   \n",
       "\n",
       "     QUAL FILTER   AC            AF  \\\n",
       "446 -10.0   PASS   28  1.841770e-04   \n",
       "447 -10.0   PASS   28  1.841770e-04   \n",
       "448 -10.0   PASS  122  8.353400e-05   \n",
       "449 -10.0   PASS  122  8.353400e-05   \n",
       "451 -10.0   PASS    1  6.847040e-07   \n",
       "..    ...    ...  ...           ...   \n",
       "580 -10.0   PASS    6  4.105090e-06   \n",
       "777 -10.0   PASS    1  6.843080e-07   \n",
       "776 -10.0   PASS    1  6.843080e-07   \n",
       "775 -10.0   PASS    3  1.970130e-05   \n",
       "774 -10.0   PASS    3  1.970130e-05   \n",
       "\n",
       "                                                   vep  index Allele  \\\n",
       "446  A|missense_variant|MODERATE|BRCA1|ENSG00000012...   6068      A   \n",
       "447  A|missense_variant|MODERATE|BRCA1|ENSG00000012...   6068      A   \n",
       "448  A|missense_variant|MODERATE|BRCA1|ENSG00000012...   6069      A   \n",
       "449  A|missense_variant|MODERATE|BRCA1|ENSG00000012...   6069      A   \n",
       "451  T|missense_variant|MODERATE|BRCA1|ENSG00000012...   6070      T   \n",
       "..                                                 ...    ...    ...   \n",
       "580  T|missense_variant|MODERATE|BRCA1|ENSG00000012...  16841      T   \n",
       "777  T|missense_variant|MODERATE|BRCA1|ENSG00000012...  21127      T   \n",
       "776  T|missense_variant|MODERATE|BRCA1|ENSG00000012...  21127      T   \n",
       "775  T|missense_variant|MODERATE|BRCA1|ENSG00000012...  21126      T   \n",
       "774  T|missense_variant|MODERATE|BRCA1|ENSG00000012...  21126      T   \n",
       "\n",
       "          Consequence    IMPACT SYMBOL             Gene Feature_type  \\\n",
       "446  missense_variant  MODERATE  BRCA1  ENSG00000012048   Transcript   \n",
       "447  missense_variant  MODERATE  BRCA1              672   Transcript   \n",
       "448  missense_variant  MODERATE  BRCA1  ENSG00000012048   Transcript   \n",
       "449  missense_variant  MODERATE  BRCA1              672   Transcript   \n",
       "451  missense_variant  MODERATE  BRCA1              672   Transcript   \n",
       "..                ...       ...    ...              ...          ...   \n",
       "580  missense_variant  MODERATE  BRCA1  ENSG00000012048   Transcript   \n",
       "777  missense_variant  MODERATE  BRCA1              672   Transcript   \n",
       "776  missense_variant  MODERATE  BRCA1  ENSG00000012048   Transcript   \n",
       "775  missense_variant  MODERATE  BRCA1              672   Transcript   \n",
       "774  missense_variant  MODERATE  BRCA1  ENSG00000012048   Transcript   \n",
       "\n",
       "             Feature         BIOTYPE   EXON  INTRON  \\\n",
       "446  ENST00000357654  protein_coding  16/23     NaN   \n",
       "447      NM_007294.4  protein_coding  16/23     NaN   \n",
       "448  ENST00000357654  protein_coding  16/23     NaN   \n",
       "449      NM_007294.4  protein_coding  16/23     NaN   \n",
       "451      NM_007294.4  protein_coding  16/23     NaN   \n",
       "..               ...             ...    ...     ...   \n",
       "580  ENST00000357654  protein_coding   5/23     NaN   \n",
       "777      NM_007294.4  protein_coding   2/23     NaN   \n",
       "776  ENST00000357654  protein_coding   2/23     NaN   \n",
       "775      NM_007294.4  protein_coding   2/23     NaN   \n",
       "774  ENST00000357654  protein_coding   2/23     NaN   \n",
       "\n",
       "                           HGVSc                           HGVSp  \\\n",
       "446  ENST00000357654.9:c.5005G>T  ENSP00000350283.3:p.Ala1669Ser   \n",
       "447        NM_007294.4:c.5005G>T        NP_009225.1:p.Ala1669Ser   \n",
       "448  ENST00000357654.9:c.5005G>T  ENSP00000350283.3:p.Ala1669Ser   \n",
       "449        NM_007294.4:c.5005G>T        NP_009225.1:p.Ala1669Ser   \n",
       "451        NM_007294.4:c.5005G>A        NP_009225.1:p.Ala1669Thr   \n",
       "..                           ...                             ...   \n",
       "580   ENST00000357654.9:c.247G>A    ENSP00000350283.3:p.Val83Ile   \n",
       "777          NM_007294.4:c.22G>A           NP_009225.1:p.Val8Ile   \n",
       "776    ENST00000357654.9:c.22G>A     ENSP00000350283.3:p.Val8Ile   \n",
       "775          NM_007294.4:c.22G>A           NP_009225.1:p.Val8Ile   \n",
       "774    ENST00000357654.9:c.22G>A     ENSP00000350283.3:p.Val8Ile   \n",
       "\n",
       "     cDNA_position  CDS_position  Protein_position Amino_acids   Codons  \\\n",
       "446           5118          5005              1669         A/S  Gcc/Tcc   \n",
       "447           5118          5005              1669         A/S  Gcc/Tcc   \n",
       "448           5118          5005              1669         A/S  Gcc/Tcc   \n",
       "449           5118          5005              1669         A/S  Gcc/Tcc   \n",
       "451           5118          5005              1669         A/T  Gcc/Acc   \n",
       "..             ...           ...               ...         ...      ...   \n",
       "580            360           247                83         V/I  Gtt/Att   \n",
       "777            135            22                 8         V/I  Gtt/Att   \n",
       "776            135            22                 8         V/I  Gtt/Att   \n",
       "775            135            22                 8         V/I  Gtt/Att   \n",
       "774            135            22                 8         V/I  Gtt/Att   \n",
       "\n",
       "     ALLELE_NUM  DISTANCE  STRAND  FLAGS VARIANT_CLASS SYMBOL_SOURCE  \\\n",
       "446           1       NaN      -1    NaN           SNV          HGNC   \n",
       "447           1       NaN      -1    NaN           SNV    EntrezGene   \n",
       "448           1       NaN      -1    NaN           SNV          HGNC   \n",
       "449           1       NaN      -1    NaN           SNV    EntrezGene   \n",
       "451           1       NaN      -1    NaN           SNV    EntrezGene   \n",
       "..          ...       ...     ...    ...           ...           ...   \n",
       "580           1       NaN      -1    NaN           SNV          HGNC   \n",
       "777           1       NaN      -1    NaN           SNV    EntrezGene   \n",
       "776           1       NaN      -1    NaN           SNV          HGNC   \n",
       "775           1       NaN      -1    NaN           SNV    EntrezGene   \n",
       "774           1       NaN      -1    NaN           SNV          HGNC   \n",
       "\n",
       "       HGNC_ID CANONICAL        MANE_SELECT  MANE_PLUS_CLINICAL  TSL APPRIS  \\\n",
       "446  HGNC:1100       YES        NM_007294.4                 NaN  1.0     A2   \n",
       "447  HGNC:1100       YES  ENST00000357654.9                 NaN  NaN    NaN   \n",
       "448  HGNC:1100       YES        NM_007294.4                 NaN  1.0     A2   \n",
       "449  HGNC:1100       YES  ENST00000357654.9                 NaN  NaN    NaN   \n",
       "451  HGNC:1100       YES  ENST00000357654.9                 NaN  NaN    NaN   \n",
       "..         ...       ...                ...                 ...  ...    ...   \n",
       "580  HGNC:1100       YES        NM_007294.4                 NaN  1.0     A2   \n",
       "777  HGNC:1100       YES  ENST00000357654.9                 NaN  NaN    NaN   \n",
       "776  HGNC:1100       YES        NM_007294.4                 NaN  1.0     A2   \n",
       "775  HGNC:1100       YES  ENST00000357654.9                 NaN  NaN    NaN   \n",
       "774  HGNC:1100       YES        NM_007294.4                 NaN  1.0     A2   \n",
       "\n",
       "            CCDS             ENSP UNIPROT_ISOFORM   SOURCE  \\\n",
       "446  CCDS11453.1  ENSP00000350283        P38398-1  Ensembl   \n",
       "447          NaN      NP_009225.1             NaN   RefSeq   \n",
       "448  CCDS11453.1  ENSP00000350283        P38398-1  Ensembl   \n",
       "449          NaN      NP_009225.1             NaN   RefSeq   \n",
       "451          NaN      NP_009225.1             NaN   RefSeq   \n",
       "..           ...              ...             ...      ...   \n",
       "580  CCDS11453.1  ENSP00000350283        P38398-1  Ensembl   \n",
       "777          NaN      NP_009225.1             NaN   RefSeq   \n",
       "776  CCDS11453.1  ENSP00000350283        P38398-1  Ensembl   \n",
       "775          NaN      NP_009225.1             NaN   RefSeq   \n",
       "774  CCDS11453.1  ENSP00000350283        P38398-1  Ensembl   \n",
       "\n",
       "                                               DOMAINS  miRNA  HGVS_OFFSET  \\\n",
       "446  ENSP_mappings:1jnx&ENSP_mappings:1n5o&ENSP_map...    NaN          NaN   \n",
       "447                                                NaN    NaN          NaN   \n",
       "448  ENSP_mappings:1jnx&ENSP_mappings:1n5o&ENSP_map...    NaN          NaN   \n",
       "449                                                NaN    NaN          NaN   \n",
       "451                                                NaN    NaN          NaN   \n",
       "..                                                 ...    ...          ...   \n",
       "580  ENSP_mappings:1jm7&Gene3D:3&PIRSF:PIRSF001734&...    NaN          NaN   \n",
       "777                                                NaN    NaN          NaN   \n",
       "776  ENSP_mappings:1jm7&Gene3D:3&PIRSF:PIRSF001734&...    NaN          NaN   \n",
       "775                                                NaN    NaN          NaN   \n",
       "774  ENSP_mappings:1jm7&Gene3D:3&PIRSF:PIRSF001734&...    NaN          NaN   \n",
       "\n",
       "     PUBMED  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS  MOTIF_SCORE_CHANGE  \\\n",
       "446     NaN         NaN        NaN           NaN                 NaN   \n",
       "447     NaN         NaN        NaN           NaN                 NaN   \n",
       "448     NaN         NaN        NaN           NaN                 NaN   \n",
       "449     NaN         NaN        NaN           NaN                 NaN   \n",
       "451     NaN         NaN        NaN           NaN                 NaN   \n",
       "..      ...         ...        ...           ...                 ...   \n",
       "580     NaN         NaN        NaN           NaN                 NaN   \n",
       "777     NaN         NaN        NaN           NaN                 NaN   \n",
       "776     NaN         NaN        NaN           NaN                 NaN   \n",
       "775     NaN         NaN        NaN           NaN                 NaN   \n",
       "774     NaN         NaN        NaN           NaN                 NaN   \n",
       "\n",
       "     TRANSCRIPTION_FACTORS  LoF  LoF_filter  LoF_flags  LoF_info  \\\n",
       "446                    NaN  NaN         NaN        NaN       NaN   \n",
       "447                    NaN  NaN         NaN        NaN       NaN   \n",
       "448                    NaN  NaN         NaN        NaN       NaN   \n",
       "449                    NaN  NaN         NaN        NaN       NaN   \n",
       "451                    NaN  NaN         NaN        NaN       NaN   \n",
       "..                     ...  ...         ...        ...       ...   \n",
       "580                    NaN  NaN         NaN        NaN       NaN   \n",
       "777                    NaN  NaN         NaN        NaN       NaN   \n",
       "776                    NaN  NaN         NaN        NaN       NaN   \n",
       "775                    NaN  NaN         NaN        NaN       NaN   \n",
       "774                    NaN  NaN         NaN        NaN       NaN   \n",
       "\n",
       "     spliceAI_scores     score  \n",
       "446             0.03 -0.401851  \n",
       "447             0.03 -0.401851  \n",
       "448             0.03 -0.401851  \n",
       "449             0.03 -0.401851  \n",
       "451             0.03  0.033971  \n",
       "..               ...       ...  \n",
       "580             0.07 -0.093432  \n",
       "777             0.02 -0.560919  \n",
       "776             0.02 -0.560919  \n",
       "775             0.02 -0.560919  \n",
       "774             0.02 -0.560919  \n",
       "\n",
       "[804 rows x 61 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomAD_ds[gnomAD_ds.duplicated(subset=[\"hgvs_pro\"],keep=False)].sort_values(\"hgvs_pro\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Case by case identify discrepancies between this pipeline and the process_data pipeline for Findlay BRCA1 SGE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "from main import load_data\n",
    "findlay_data = load_data(data_directory=Path(\"/data/dzeiberg/mave_calibration_debug_data\"),dataset_id=\"Findlay_BRCA1_SGE\")\n",
    "findlay_observations = joblib.load(Path(\"/data/dzeiberg/mave_calibration_debug_data\") / \"Findlay_BRCA1_SGE\" / \"hgvs_pro.pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(findlay_observations['p_lp']) - set(clinvar_ds.loc[clinvar_ds.is_pathogenic,'hgvs_pro'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(findlay_observations['b_lb']) - set(clinvar_ds.loc[clinvar_ds.is_benign,'hgvs_pro'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_from_gnomAD = set(findlay_observations['gnomad']) - set(gnomAD_ds.hgvs_pro.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq</th>\n",
       "      <th>MANE_RefSeq_nuc</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>MANE_RefSeq_prot</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>ENSEMBL_nuc</th>\n",
       "      <th>ENSEMBL_prot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P38398</th>\n",
       "      <td>MDLSALRVEEVQNVINAMQKILECPICLELIKEPVSTKCDHIFCKF...</td>\n",
       "      <td>NM_007294.4</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>NP_009225.1</td>\n",
       "      <td>17</td>\n",
       "      <td>43044295</td>\n",
       "      <td>43170245</td>\n",
       "      <td>HGNC:1100</td>\n",
       "      <td>-</td>\n",
       "      <td>672</td>\n",
       "      <td>ENST00000357654.9</td>\n",
       "      <td>ENSP00000350283.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P04637</th>\n",
       "      <td>MEEPQSDPSVEPPLSQETFSDLWKLLPENNVLSPLPSQAMDDLMLS...</td>\n",
       "      <td>NM_000546.6</td>\n",
       "      <td>TP53</td>\n",
       "      <td>NP_000537.3</td>\n",
       "      <td>17</td>\n",
       "      <td>7661779</td>\n",
       "      <td>7687546</td>\n",
       "      <td>HGNC:11998</td>\n",
       "      <td>-</td>\n",
       "      <td>7157</td>\n",
       "      <td>ENST00000269305.9</td>\n",
       "      <td>ENSP00000269305.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P51587</th>\n",
       "      <td>MPIGSKERPTFFEIFKTRCNKADLGPISLNWFEELSSEAPPYNSEP...</td>\n",
       "      <td>NM_000059.4</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>NP_000050.3</td>\n",
       "      <td>13</td>\n",
       "      <td>32315086</td>\n",
       "      <td>32400268</td>\n",
       "      <td>HGNC:1101</td>\n",
       "      <td>+</td>\n",
       "      <td>675</td>\n",
       "      <td>ENST00000380152.8</td>\n",
       "      <td>ENSP00000369497.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>O15118</th>\n",
       "      <td>MTARGLALGLLLLLLCPAQVFSQSCVWYGECGIAYGDKRYNCEYSG...</td>\n",
       "      <td>NM_000271.5</td>\n",
       "      <td>NPC1</td>\n",
       "      <td>NP_000262.2</td>\n",
       "      <td>18</td>\n",
       "      <td>23506184</td>\n",
       "      <td>23586506</td>\n",
       "      <td>HGNC:7897</td>\n",
       "      <td>-</td>\n",
       "      <td>4864</td>\n",
       "      <td>ENST00000269228.10</td>\n",
       "      <td>ENSP00000269228.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P40337</th>\n",
       "      <td>MPRRAENWDEAEVGAEEAGVEEYGPEEDGGEESGAEESGPEESGPE...</td>\n",
       "      <td>NM_000551.4</td>\n",
       "      <td>VHL</td>\n",
       "      <td>NP_000542.1</td>\n",
       "      <td>3</td>\n",
       "      <td>10141778</td>\n",
       "      <td>10153667</td>\n",
       "      <td>HGNC:12687</td>\n",
       "      <td>+</td>\n",
       "      <td>7428</td>\n",
       "      <td>ENST00000256474.3</td>\n",
       "      <td>ENSP00000256474.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>O96017</th>\n",
       "      <td>MSRESDVEAQQSHGSSACSQPHGSVTQSQGSSSQSQGISSSSTSTM...</td>\n",
       "      <td>NM_007194.4</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>NP_009125.1</td>\n",
       "      <td>22</td>\n",
       "      <td>28687743</td>\n",
       "      <td>28742422</td>\n",
       "      <td>HGNC:16627</td>\n",
       "      <td>-</td>\n",
       "      <td>11200</td>\n",
       "      <td>ENST00000404276.6</td>\n",
       "      <td>ENSP00000385747.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P43246</th>\n",
       "      <td>MAVQPKETLQLESAAEVGFVRFFQGMPEKPTTTVRLFDRGDFYTAH...</td>\n",
       "      <td>NM_000251.3</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>NP_000242.1</td>\n",
       "      <td>2</td>\n",
       "      <td>47403067</td>\n",
       "      <td>47663146</td>\n",
       "      <td>HGNC:7325</td>\n",
       "      <td>+</td>\n",
       "      <td>4436</td>\n",
       "      <td>ENST00000233146.7</td>\n",
       "      <td>ENSP00000233146.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P60484</th>\n",
       "      <td>MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVY...</td>\n",
       "      <td>NM_000314.8</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>NP_000305.3</td>\n",
       "      <td>10</td>\n",
       "      <td>87862638</td>\n",
       "      <td>87971930</td>\n",
       "      <td>HGNC:9588</td>\n",
       "      <td>+</td>\n",
       "      <td>5728</td>\n",
       "      <td>ENST00000371953.8</td>\n",
       "      <td>ENSP00000361021.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P51580</th>\n",
       "      <td>MDGTRTSLDIEEYSDTEVQKNQVLTLEEWQDKWVNGKTAFHQEQGH...</td>\n",
       "      <td>NM_000367.5</td>\n",
       "      <td>TPMT</td>\n",
       "      <td>NP_000358.1</td>\n",
       "      <td>6</td>\n",
       "      <td>18128311</td>\n",
       "      <td>18155077</td>\n",
       "      <td>HGNC:12014</td>\n",
       "      <td>-</td>\n",
       "      <td>7172</td>\n",
       "      <td>ENST00000309983.5</td>\n",
       "      <td>ENSP00000312304.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>O00571</th>\n",
       "      <td>MSHVAVENALGLDQQFAGLDLNSSDNQSGGSTASKGRYIPPHLRNR...</td>\n",
       "      <td>NM_001356.5</td>\n",
       "      <td>DDX3X</td>\n",
       "      <td>NP_001347.3</td>\n",
       "      <td>X</td>\n",
       "      <td>41333348</td>\n",
       "      <td>41364472</td>\n",
       "      <td>HGNC:2745</td>\n",
       "      <td>+</td>\n",
       "      <td>1654</td>\n",
       "      <td>ENST00000644876.2</td>\n",
       "      <td>ENSP00000494040.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P08397</th>\n",
       "      <td>MSGNGNAAATAEENSPKMRVIRVGTRKSQLARIQTDSVVATLKASY...</td>\n",
       "      <td>NM_000190.4</td>\n",
       "      <td>HMBS</td>\n",
       "      <td>NP_000181.2</td>\n",
       "      <td>11</td>\n",
       "      <td>119084866</td>\n",
       "      <td>119093834</td>\n",
       "      <td>HGNC:4982</td>\n",
       "      <td>+</td>\n",
       "      <td>3145</td>\n",
       "      <td>ENST00000652429.1</td>\n",
       "      <td>ENSP00000498786.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      seq MANE_RefSeq_nuc  \\\n",
       "P38398  MDLSALRVEEVQNVINAMQKILECPICLELIKEPVSTKCDHIFCKF...     NM_007294.4   \n",
       "P04637  MEEPQSDPSVEPPLSQETFSDLWKLLPENNVLSPLPSQAMDDLMLS...     NM_000546.6   \n",
       "P51587  MPIGSKERPTFFEIFKTRCNKADLGPISLNWFEELSSEAPPYNSEP...     NM_000059.4   \n",
       "O15118  MTARGLALGLLLLLLCPAQVFSQSCVWYGECGIAYGDKRYNCEYSG...     NM_000271.5   \n",
       "P40337  MPRRAENWDEAEVGAEEAGVEEYGPEEDGGEESGAEESGPEESGPE...     NM_000551.4   \n",
       "O96017  MSRESDVEAQQSHGSSACSQPHGSVTQSQGSSSQSQGISSSSTSTM...     NM_007194.4   \n",
       "P43246  MAVQPKETLQLESAAEVGFVRFFQGMPEKPTTTVRLFDRGDFYTAH...     NM_000251.3   \n",
       "P60484  MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVY...     NM_000314.8   \n",
       "P51580  MDGTRTSLDIEEYSDTEVQKNQVLTLEEWQDKWVNGKTAFHQEQGH...     NM_000367.5   \n",
       "O00571  MSHVAVENALGLDQQFAGLDLNSSDNQSGGSTASKGRYIPPHLRNR...     NM_001356.5   \n",
       "P08397  MSGNGNAAATAEENSPKMRVIRVGTRKSQLARIQTDSVVATLKASY...     NM_000190.4   \n",
       "\n",
       "       gene_name MANE_RefSeq_prot CHROM      START       STOP     HGNC_ID  \\\n",
       "P38398     BRCA1      NP_009225.1    17   43044295   43170245   HGNC:1100   \n",
       "P04637      TP53      NP_000537.3    17    7661779    7687546  HGNC:11998   \n",
       "P51587     BRCA2      NP_000050.3    13   32315086   32400268   HGNC:1101   \n",
       "O15118      NPC1      NP_000262.2    18   23506184   23586506   HGNC:7897   \n",
       "P40337       VHL      NP_000542.1     3   10141778   10153667  HGNC:12687   \n",
       "O96017     CHEK2      NP_009125.1    22   28687743   28742422  HGNC:16627   \n",
       "P43246      MSH2      NP_000242.1     2   47403067   47663146   HGNC:7325   \n",
       "P60484      PTEN      NP_000305.3    10   87862638   87971930   HGNC:9588   \n",
       "P51580      TPMT      NP_000358.1     6   18128311   18155077  HGNC:12014   \n",
       "O00571     DDX3X      NP_001347.3     X   41333348   41364472   HGNC:2745   \n",
       "P08397      HMBS      NP_000181.2    11  119084866  119093834   HGNC:4982   \n",
       "\n",
       "       STRAND GeneID         ENSEMBL_nuc       ENSEMBL_prot  \n",
       "P38398      -    672   ENST00000357654.9  ENSP00000350283.3  \n",
       "P04637      -   7157   ENST00000269305.9  ENSP00000269305.4  \n",
       "P51587      +    675   ENST00000380152.8  ENSP00000369497.3  \n",
       "O15118      -   4864  ENST00000269228.10  ENSP00000269228.4  \n",
       "P40337      +   7428   ENST00000256474.3  ENSP00000256474.3  \n",
       "O96017      -  11200   ENST00000404276.6  ENSP00000385747.1  \n",
       "P43246      +   4436   ENST00000233146.7  ENSP00000233146.2  \n",
       "P60484      +   5728   ENST00000371953.8  ENSP00000361021.3  \n",
       "P51580      -   7172   ENST00000309983.5  ENSP00000312304.4  \n",
       "O00571      +   1654   ENST00000644876.2  ENSP00000494040.1  \n",
       "P08397      +   3145   ENST00000652429.1  ENSP00000498786.1  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ENST00000357654', 'NM_007294'}"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(gene_info.loc[metadata.loc['Findlay_BRCA1_SGE','uniprot_acc'],['MANE_RefSeq_nuc','ENSEMBL_nuc']].apply(lambda x: x.split('.')[0]).values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ENST00000493795', 'NM_007297.4', 'ENST00000591849',\n",
       "       'ENST00000586385', 'ENST00000471181', 'NM_007300.4',\n",
       "       'ENST00000461574', 'ENST00000644379', 'ENST00000497488',\n",
       "       'ENST00000412061', 'ENST00000473961', 'ENST00000484087',\n",
       "       'ENST00000487825', 'ENST00000493919', 'ENST00000652672',\n",
       "       'ENST00000477152', 'ENST00000489037', 'ENST00000644555'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomad_df[(gnomad_df.hgvs_pro.isin(missing_from_gnomAD)) & (gnomad_df.HGNC_ID == gene_info.loc[metadata.loc['Findlay_BRCA1_SGE','uniprot_acc'],'HGNC_ID'])].Feature.unique()# & (gnomad_df.Feature_base.isin(set(gene_info.loc[metadata.loc['Findlay_BRCA1_SGE','uniprot_acc'],['MANE_RefSeq_nuc','ENSEMBL_nuc']].apply(lambda x: x.split('.')[0]).values)))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_sample(scores,ax=None,label=\"\"):\n",
    "    if ax is None:\n",
    "        fig,ax = plt.subplots()\n",
    "    sns.histplot(scores,label=f\"{label} (n={len(scores)})\",ax=ax,stat='density',bins=15)\n",
    "    ax.legend()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for dataset_name, scoreset in tqdm(scoresets.items(),total=len(scoresets)):\n",
    "    gnomAD_ds = get_gnomAD_subset(scoresets[dataset_name])\n",
    "    clinvar_ds = get_clinvar_subset(scoresets[dataset_name])\n",
    "    savedir = data_directory / dataset_name / \"processed\"\n",
    "    savedir.mkdir(exist_ok=True,parents=True)\n",
    "    gnomAD_ds.to_csv(savedir / \"gnomAD.csv\")\n",
    "    clinvar_ds.to_csv(savedir / \"ClinVar.csv\")\n",
    "    fig,ax = plt.subplots(4,1,figsize=(10,12),sharex=True,sharey=True)\n",
    "    plot_sample(clinvar_ds[(clinvar_ds.is_pathogenic) & \\\n",
    "                    (clinvar_ds.snv_category == \"missense\") & \\\n",
    "                        (clinvar_ds.spliceAI_score < .5)].drop_duplicates('Name').score.values,ax=ax[0],label=\"P/LP\")\n",
    "\n",
    "    plot_sample(clinvar_ds[(clinvar_ds.is_benign) & \\\n",
    "                    (clinvar_ds.snv_category == \"missense\") & \\\n",
    "                        (clinvar_ds.spliceAI_score < .5)].drop_duplicates('Name').score.values,ax=ax[1],label=\"B/LB\")\n",
    "    plot_sample(gnomAD_ds.score,ax=ax[2],label=\"gnomAD\")\n",
    "    clinvar_ds = clinvar_ds.reset_index()\n",
    "    silent_pathogenic_variants = set(clinvar_ds.loc[(clinvar_ds.snv_category == \"silent\") & \\\n",
    "                                                    (clinvar_ds.is_pathogenic),'hgvs_pro'].values)\n",
    "    plot_sample(scoreset[(scoreset.synonymous) & \\\n",
    "                            (~scoreset.hgvs_pro.isin(silent_pathogenic_variants))].score,ax=ax[3],label=\"Synonymous\")\n",
    "    fig.savefig(savedir / \"score_distributions.jpeg\",dpi=300,bbox_inches='tight')\n",
    "    plt.close(fig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnomad_df.loc[(gnomad_df.HGNC_ID == gene_info.loc['P38398','HGNC_ID']) & \\\n",
    "                (gnomad_df.hgvs_pro.isin(set(scoresets['Findlay_BRCA1_SGE'].hgvs_pro.values))) &\\\n",
    "                    ((gnomad_df.spliceAI_scores.isna()) | (gnomad_df.spliceAI_scores < .5)) & \\\n",
    "                        (~gnomad_df.hgvs_pro.isin(set(gnomAD_ds.hgvs_pro.values)))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_clinvar_subset??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_df.loc[(clinvar_df.transcript == gene_info.loc['P38398','MANE_RefSeq_nuc']) & \\\n",
    "                (clinvar_df.hgvs_pro.isin(set(scoresets['Findlay_BRCA1_SGE'].hgvs_pro.values))) &\\\n",
    "                (clinvar_df.is_pathogenic) & \\\n",
    "                    (clinvar_df.snv_category == \"missense\") & \\\n",
    "                    ((clinvar_df.spliceAI_score.isna()) | (clinvar_df.spliceAI_score < .5))]# & \\\n",
    "                        #(~clinvar_df.hgvs_pro.isin(set(clinvar_ds.reset_index().loc[clinvar_ds.is_pathogenic].hgvs_pro.values)))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(set(findlay_observations['b_lb']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_df.loc[(clinvar_df.hgvs_pro.isin(set(findlay_observations['b_lb']))) & \\\n",
    "                (clinvar_df.transcript == gene_info.loc['P38398','MANE_RefSeq_nuc']) & \\\n",
    "                    ((clinvar_df.spliceAI_score.isna()) | (clinvar_df.spliceAI_score < .5)) & \\\n",
    "                        (clinvar_df.hgvs_pro.isin(set(scoresets['Findlay_BRCA1_SGE'].hgvs_pro.values))) & \\\n",
    "                    (~clinvar_df.hgvs_pro.isin(set(clinvar_ds.loc[(clinvar_ds.is_benign) & \\\n",
    "                                                                    (clinvar_ds.snv_category == \"missense\") & \\\n",
    "                                                                        ((clinvar_ds.spliceAI_score < .5) | \\\n",
    "                                                                            (clinvar_ds.spliceAI_score.isna()))].hgvs_pro.values)))].drop_duplicates('hgvs_pro')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_ds.loc[(clinvar_ds.is_pathogenic) & \\\n",
    "                (clinvar_ds.snv_category == \"missense\") & \\\n",
    "                    ((clinvar_ds.spliceAI_score < .5) | \\\n",
    "                        (clinvar_ds.spliceAI_score.isna()))].drop_duplicates('hgvs_pro').shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_ds.loc[(clinvar_ds.is_benign) & \\\n",
    "                (clinvar_ds.snv_category == \"missense\") & \\\n",
    "                    ((clinvar_ds.spliceAI_score < .5) | \\\n",
    "                        (clinvar_ds.spliceAI_score.isna()))].drop_duplicates('hgvs_pro').shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "results_dir = Path(\"/data/dzeiberg/mave_calibration/results_08_26_24/figs3\")\n",
    "scoreset_thresholds = {}\n",
    "for fp in results_dir.glob(\"*.json\"):\n",
    "    with open(fp) as f:\n",
    "        summary = json.load(f)\n",
    "    scoreset_thresholds[os.path.basename(fp).replace(\".json\",\"\")] = dict(p_lp=summary['pathogenic_score_thresholds'],b_lb=summary['benign_score_thresholds'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def assign_assay_evidence_strength(score, thresholds):\n",
    "    if np.isnan(score):\n",
    "        return 0\n",
    "    for threshold,points in list(zip(thresholds[\"p_lp\"],[1,2,3,4,8]))[::-1]:\n",
    "        if np.isnan(threshold):\n",
    "            continue\n",
    "        if score <= threshold:\n",
    "            return points\n",
    "    for threshold,points in list(zip(thresholds[\"b_lb\"],[-1,-2,-3,-4,-8]))[::-1]:\n",
    "        if np.isnan(threshold):\n",
    "            continue\n",
    "        if score >= threshold:\n",
    "            return points\n",
    "    return 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "scoreset[scoreset.hgvs_pro == \"p.Ala1823Pro\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_scored = pd.merge(clinvar_df,scoreset_df,\n",
    "            left_on=['hgvs_pro','transcript'],\n",
    "            right_on=['hgvs_pro','MANE_RefSeq_nuc'],\n",
    "            how='left')\n",
    "gnomad_scored = pd.merge(gnomad_df,scoreset_df,\n",
    "            left_on=['hgvs_pro','HGNC_ID'],\n",
    "            right_on=['hgvs_pro','HGNC_ID'],\n",
    "            how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_scored[(clinvar_scored.hgvs_pro == \"p.Ala1823Pro\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "f\"{clinvar_scored.shape[0]} ClinVar score records (1 record per score)\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "f\"{gnomad_scored.shape[0]} gnomAD score records (1 record per score)\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "metadata": {},
   "outputs": [],
   "source": [
    "# assay_cols = [*[f\"score_{k}\" for k in scoreset_thresholds.keys()],*[f\"strength_{k}\" for k in scoreset_thresholds.keys()]]\n",
    "assay_cols = ['score_Findlay_BRCA1_SGE','strength_Findlay_BRCA1_SGE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_assays_aggregated = pd.merge(clinvar_scored.loc[:,list(set(clinvar_scored.columns) - set(assay_cols))].drop_duplicates(subset=['hgvs_pro','transcript']),\n",
    "                                    clinvar_scored.loc[:,['hgvs_pro','transcript',*assay_cols]].groupby(['hgvs_pro','transcript'], as_index=True).agg(np.nanmax),\n",
    "                                    on=['hgvs_pro','transcript'],)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnomad_assays_aggregated = pd.merge(gnomad_scored.loc[:,list(set(gnomad_scored.columns) - set(assay_cols))].drop_duplicates(subset=['hgvs_pro','HGNC_ID']),\n",
    "                                    gnomad_scored.loc[:,['hgvs_pro','HGNC_ID',*assay_cols]].groupby(['hgvs_pro','HGNC_ID'], as_index=True).agg(np.nanmax),\n",
    "                                    on=['hgvs_pro','HGNC_ID'],)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnomad_assays_aggregated = gnomad_assays_aggregated.assign(CHROM=gnomad_assays_aggregated.CHROM.str.replace('chr',''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_altering_variants = pd.concat([pd.read_csv(f) for f in data_directory.glob(\"**/*spliceAI_filtered.csv\")])\n",
    "splice_altering_variants = splice_altering_variants.assign(POS=splice_altering_variants.POS.astype(str),\n",
    "                                                            CHROM=splice_altering_variants.CHROM.astype(str))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_assays_aggregated_wo_splice = clinvar_assays_aggregated[~clinvar_assays_aggregated.Name.isin(set(splice_altering_variants.Name.values))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_assays_aggregated.shape[0], clinvar_assays_aggregated_wo_splice.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_set = set(splice_altering_variants.loc[:,[\"CHROM\",\"POS\",\"REF\",\"ALT\"]].astype(str).apply(tuple,axis=1).values)\n",
    "gnomad_assays_aggregated_wo_splice = gnomad_assays_aggregated[gnomad_assays_aggregated.loc[:,['CHROM','POS','REF','ALT']].astype(str).apply(tuple, axis=1).apply(lambda x: x not in splice_set)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnomad_assays_aggregated.shape[0], gnomad_assays_aggregated_wo_splice.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "color_palette = sns.choose_diverging_palette()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# strength_col = \"strength_Findlay_BRCA1_SGE\"\n",
    "Normalize_counts = True\n",
    "strength_range = np.array([-8,-4,-3,-2,-1,0,1,2,3,4,8]).astype(float)\n",
    "color = {-8:color_palette[0],-4:color_palette[1],-3:color_palette[2],-2:color_palette[3],-1:color_palette[4],\n",
    "        0:color_palette[5],1:color_palette[6],2:color_palette[7],3:color_palette[8],4:color_palette[9],8:color_palette[10]}\n",
    "# for strength_col in df_scoreset.columns[df_scoreset.columns.str.startswith(\"strength_\")]:    \n",
    "# for dataset_name in scoreset_thresholds.keys():\n",
    "for dataset_name in [\"Findlay_BRCA1_SGE\",]:\n",
    "    masks = {\"P/LP\":(~clinvar_assays_aggregated_wo_splice.loc[:,f\"strength_{dataset_name}\"].isna()) & clinvar_assays_aggregated_wo_splice.is_pathogenic,\n",
    "            \"B/LB\":(~clinvar_assays_aggregated_wo_splice.loc[:,f\"strength_{dataset_name}\"].isna()) & clinvar_assays_aggregated_wo_splice.is_benign,\n",
    "            \"VUS\":(~clinvar_assays_aggregated_wo_splice.loc[:,f\"strength_{dataset_name}\"].isna()) & (clinvar_assays_aggregated_wo_splice.ClinicalSignificance == \"Uncertain significance\"),\n",
    "            \"Conflicting\":(~clinvar_assays_aggregated_wo_splice.loc[:,f\"strength_{dataset_name}\"].isna()) & (clinvar_assays_aggregated_wo_splice.ClinicalSignificance == \"Conflicting classifications of pathogenicity\"),\n",
    "            \"Missense\":(~clinvar_assays_aggregated_wo_splice.loc[:,f\"strength_{dataset_name}\"].isna()) & (clinvar_assays_aggregated_wo_splice.snv_category == \"missense\")}\n",
    "            \n",
    "    strengths = {f\"{k} (n={mask.sum():,d})\": clinvar_assays_aggregated_wo_splice[mask].loc[:,f\"strength_{dataset_name}\"].value_counts(normalize=Normalize_counts).fillna(0).astype(float) for k,mask in masks.items()}\n",
    "    gnomAD_mask = ~gnomad_assays_aggregated.loc[:,f\"strength_{dataset_name}\"].isna()\n",
    "    strengths[f\"gnomAD (n={gnomAD_mask.sum():,d})\"] = gnomad_assays_aggregated.loc[gnomAD_mask,f\"strength_{dataset_name}\"].value_counts(normalize=Normalize_counts).fillna(0).astype(float)\n",
    "    strengths = pd.DataFrame.from_dict(strengths,orient='index')\n",
    "    strengths = strengths.rename(columns={s : int(s) for s in strengths.columns})\n",
    "    for s in strength_range:\n",
    "        if s not in strengths.columns:\n",
    "            strengths[s] = 0\n",
    "    strengths = strengths.loc[:,strengths.columns.sort_values()]\n",
    "    strengths.plot(kind='barh',\n",
    "                    stacked=True,\n",
    "                    color=color)\n",
    "    plt.legend(loc='center left', bbox_to_anchor=(1, 0.5))\n",
    "    fig = plt.gcf()\n",
    "    plt.title(f\"{dataset_name.replace('_',' ')} strength distribution\")\n",
    "    # fig.savefig(results_dir / f\"{dataset_name}_evidence_strengths.jpg\",bbox_inches='tight',dpi=300)\n",
    "    plt.show()\n",
    "    plt.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "metadata": {},
   "outputs": [],
   "source": [
    "expected_hgvs_pro = joblib.load(data_directory / dataset_name / \"hgvs_pro.pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(expected_hgvs_pro['p_lp'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "masks['P/LP'].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_assays_aggregated_wo_splice.loc[(masks['B/LB']) & (~clinvar_assays_aggregated_wo_splice.hgvs_pro.isin(set(expected_hgvs_pro['b_lb'].values))),'hgvs_pro']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "set(expected_hgvs_pro['b_lb'].values) - set(clinvar_assays_aggregated_wo_splice.loc[masks['B/LB'],'hgvs_pro'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar[(clinvar.hgvs_pro == \"p.Thr37=\") & (clinvar.GeneSymbol == \"BRCA1\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "scoreset[scoreset.hgvs_pro == \"p.Thr37=\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_scored[(clinvar_scored.hgvs_pro == \"p.Thr37=\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_assays_aggregated[clinvar_assays_aggregated.hgvs_pro == \"p.Thr37=\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "set(expected_hgvs_pro['p_lp'].values) - set(clinvar_assays_aggregated_wo_splice.loc[masks['P/LP'],'hgvs_pro'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_altering_variants[splice_altering_variants.Name == \"NM_007294.4(BRCA1):c.5467G>C (p.Ala1823Pro)\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mave-calibration-xCjwkSk6-py3.11",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
